core/src/runtime/mod.rs

```diff
@@ -935,7 +935,7 @@ pub mod tests {
     let ctx = AthenaContext::new(wallet_template, ADDRESS_ALICE, 0);
     let mut runtime = Runtime::new(program.clone(), Some(&mut host), opts, Some(ctx));
     stdin.write_vec(spawn_result.blob.clone());
-    stdin.write_vec(to_vec(&send_args).expect("failed to serialize wallet send arguments"));
+    stdin.write_vec(send_args.encode());
```

Comment:
out of curiosity, how does error handling happen here? does `encode()` panic on error?

Reply:
As far as I understand, encoding cannot fail (except for running out of memory when the allocator fails, when growing the Vec and Rust will panic in this scenario "by itself").